### PR TITLE
etcdserver: refuse to restart if backend file is missing

### PR DESCRIPTION
**Pull request by xiang90**
```bash
./etcd
...
ctrl+c
...
rm default.etcd/member/snap/db
...
./etcd
...
2016-06-26 20:13:02.596930 C | etcdmain: database file (default.etcd/member/snap/db) of backend is missing
```

### CODE REVIEW

File: etcdserver/server.go [hunk 1]
```diff
@@ -351,6 +354,10 @@ func NewServer(cfg *ServerConfig) (srv *EtcdServer, err error) {
 		cl.SetStore(st)
 		cl.SetBackend(be)
 		cl.Recover()
+		if cl.Version != nil && cl.Version().LessThan(semver.Version{Major: 3}) && !beExist {
+			os.RemoveAll(bepath)
```
Comment (heyitsanthony) [inline etcdserver/server.go]: this seems racey-- delete the file, bring up etcd, let the backend create a new bepath, then kill etcd before it reaches this point, start etcd again and it'll boot

Comment (gyuho) [inline etcdserver/server.go]: Why do we need this when `!beExist`? Is it just to clean up?

Comment (xiang90) [inline etcdserver/server.go]: not for cleanup. it is for checking. if the cluster version is 3, the be must exist before.

for the racey concern, this is a just a safety guard. We will check the consistent index of the beckend anyway. so even if we fail to remove it, the new backend will not contain the correct consistent index.

**Comment by gyuho** [comment]
LGTM. Thanks.

**Comment by heyitsanthony** [comment]
lgtm

**Pull request merged**